meson: Missing function declarations for text-utils

Source files in `text-utils` fail to compile due to missing function declarations. This appears to be due to a missing dependency such as tinfo or ncurses. Meson should properly handle this requirement when running the `meson setup` command.

```
 meson compile -C build
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /usr/bin/ninja -C /var/home/jordan/Projects/util-linux/build
ninja: Entering directory `/var/home/jordan/Projects/util-linux/build'
[302/757] Compiling C object ul.p/text-utils_ul.c.o
FAILED: ul.p/text-utils_ul.c.o 
cc -Iul.p -I. -I.. -Iinclude -I../include -Ilibsmartcols -I../libsmartcols -I../libsmartcols/src -Ilibfdisk -I../libfdisk -I../libfdisk/src -I../libuuid/src -I../liblastlog2/src -Isys-utils -I../sys-utils -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -O0 -g -D_GNU_SOURCE -fsigned-char -include /var/home/jordan/Projects/util-linux/build/config.h -fno-common -Waddress-of-packed-member -Wdiscarded-qualifiers -Wformat-security -Wimplicit-function-declaration -Wmissing-declarations -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-missing-field-initializers -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wsign-compare -Wstrict-prototypes -Wtype-limits -Wuninitialized -Wunused-but-set-parameter -Wunused-but-set-variable -Wunused-parameter -Wunused-result -Wunused-variable -Wvla -Walloca -MD -MQ ul.p/text-utils_ul.c.o -MF ul.p/text-utils_ul.c.o.d -o ul.p/text-utils_ul.c.o -c ../text-utils/ul.c
../text-utils/ul.c: In function ‘init_term_caps’:
../text-utils/ul.c:173:35: error: implicit declaration of function ‘tigetstr’ [-Wimplicit-function-declaration]
  173 |         tcs->curs_up            = tigetstr("cuu1");
      |                                   ^~~~~~~~
../text-utils/ul.c:173:35: warning: nested extern declaration of ‘tigetstr’ [-Wnested-externs]
../text-utils/ul.c:173:33: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  173 |         tcs->curs_up            = tigetstr("cuu1");
      |                                 ^
../text-utils/ul.c:174:33: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  174 |         tcs->curs_right         = tigetstr("cuf1");
      |                                 ^
../text-utils/ul.c:175:33: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  175 |         tcs->curs_left          = tigetstr("cub1");
      |                                 ^
../text-utils/ul.c:179:33: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  179 |         tcs->enter_standout     = tigetstr("smso");
      |                                 ^
../text-utils/ul.c:180:33: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  180 |         tcs->exit_standout      = tigetstr("rmso");
      |                                 ^
../text-utils/ul.c:181:33: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  181 |         tcs->enter_underline    = tigetstr("smul");
      |                                 ^
../text-utils/ul.c:182:33: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  182 |         tcs->exit_underline     = tigetstr("rmul");
      |                                 ^
../text-utils/ul.c:183:33: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  183 |         tcs->enter_dim          = tigetstr("dim");
      |                                 ^
../text-utils/ul.c:184:33: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  184 |         tcs->enter_bold         = tigetstr("bold");
      |                                 ^
../text-utils/ul.c:185:33: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  185 |         tcs->enter_reverse      = tigetstr("rev");
      |                                 ^
../text-utils/ul.c:186:33: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  186 |         tcs->exit_attributes    = tigetstr("sgr0");
      |                                 ^
../text-utils/ul.c:215:25: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  215 |         tcs->under_char = tigetstr("uc");
      |                         ^
../text-utils/ul.c:218:14: error: implicit declaration of function ‘tigetflag’ [-Wimplicit-function-declaration]
  218 |         if ((tigetflag("os") && tcs->enter_bold == NULL) ||
      |              ^~~~~~~~~
../text-utils/ul.c:218:14: warning: nested extern declaration of ‘tigetflag’ [-Wnested-externs]
../text-utils/ul.c: In function ‘print_line’:
../text-utils/ul.c:240:9: error: implicit declaration of function ‘tputs’; did you mean ‘puts’? [-Wimplicit-function-declaration]
  240 |         tputs(line, STDOUT_FILENO, ul_putwchar);
      |         ^~~~~
      |         puts
../text-utils/ul.c:240:9: warning: nested extern declaration of ‘tputs’ [-Wnested-externs]
../text-utils/ul.c: In function ‘main’:
../text-utils/ul.c:613:9: error: implicit declaration of function ‘setupterm’ [-Wimplicit-function-declaration]
  613 |         setupterm(termtype, STDOUT_FILENO, &ret);
      |         ^~~~~~~~~
../text-utils/ul.c:613:9: warning: nested extern declaration of ‘setupterm’ [-Wnested-externs]
../text-utils/ul.c:644:9: error: implicit declaration of function ‘del_curterm’ [-Wimplicit-function-declaration]
  644 |         del_curterm(cur_term);
      |         ^~~~~~~~~~~
../text-utils/ul.c:644:9: warning: nested extern declaration of ‘del_curterm’ [-Wnested-externs]
../text-utils/ul.c:644:21: error: ‘cur_term’ undeclared (first use in this function)
  644 |         del_curterm(cur_term);
      |                     ^~~~~~~~
../text-utils/ul.c:644:21: note: each undeclared identifier is reported only once for each function it appears in
[313/757] Compiling C object pg.p/text-utils_pg.c.o
FAILED: pg.p/text-utils_pg.c.o 
cc -Ipg.p -I. -I.. -Iinclude -I../include -Ilibsmartcols -I../libsmartcols -I../libsmartcols/src -Ilibfdisk -I../libfdisk -I../libfdisk/src -I../libuuid/src -I../liblastlog2/src -Isys-utils -I../sys-utils -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -O0 -g -D_GNU_SOURCE -fsigned-char -include /var/home/jordan/Projects/util-linux/build/config.h -fno-common -Waddress-of-packed-member -Wdiscarded-qualifiers -Wformat-security -Wimplicit-function-declaration -Wmissing-declarations -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-missing-field-initializers -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wsign-compare -Wstrict-prototypes -Wtype-limits -Wuninitialized -Wunused-but-set-parameter -Wunused-but-set-variable -Wunused-parameter -Wunused-result -Wunused-variable -Wvla -Walloca -MD -MQ pg.p/text-utils_pg.c.o -MF pg.p/text-utils_pg.c.o.d -o pg.p/text-utils_pg.c.o -c ../text-utils/pg.c
../text-utils/pg.c: In function ‘mesg’:
../text-utils/pg.c:306:17: error: implicit declaration of function ‘vidputs’; did you mean ‘puts’? [-Wimplicit-function-declaration]
  306 |                 vidputs(A_STANDOUT, outcap);
      |                 ^~~~~~~
      |                 puts
../text-utils/pg.c:306:17: warning: nested extern declaration of ‘vidputs’ [-Wnested-externs]
../text-utils/pg.c:306:25: error: ‘A_STANDOUT’ undeclared (first use in this function)
  306 |                 vidputs(A_STANDOUT, outcap);
      |                         ^~~~~~~~~~
../text-utils/pg.c:306:25: note: each undeclared identifier is reported only once for each function it appears in
../text-utils/pg.c:309:25: error: ‘A_NORMAL’ undeclared (first use in this function)
  309 |                 vidputs(A_NORMAL, outcap);
      |                         ^~~~~~~~
../text-utils/pg.c: In function ‘getwinsize’:
../text-utils/pg.c:329:39: error: ‘columns’ undeclared (first use in this function)
  329 |                 if (tinfostat != 1 || columns == 0)
      |                                       ^~~~~~~
../text-utils/pg.c:333:39: error: ‘lines’ undeclared (first use in this function)
  333 |                 if (tinfostat != 1 || lines == 0)
      |                                       ^~~~~
../text-utils/pg.c: In function ‘getstate’:
../text-utils/pg.c:570:21: error: ‘bell’ undeclared (first use in this function)
  570 |                 if (bell)
      |                     ^~~~
../text-utils/pg.c:571:25: error: implicit declaration of function ‘tputs’; did you mean ‘puts’? [-Wimplicit-function-declaration]
  571 |                         tputs(bell, STDOUT_FILENO, outcap);
      |                         ^~~~~
      |                         puts
../text-utils/pg.c:571:25: warning: nested extern declaration of ‘tputs’ [-Wnested-externs]
../text-utils/pg.c: In function ‘pgfile’:
../text-utils/pg.c:1064:38: error: ‘clear_screen’ undeclared (first use in this function)
 1064 |                         if (cflag && clear_screen) {
      |                                      ^~~~~~~~~~~~
../text-utils/pg.c: In function ‘main’:
../text-utils/pg.c:1557:17: error: implicit declaration of function ‘setupterm’ [-Wimplicit-function-declaration]
 1557 |                 setupterm(NULL, STDOUT_FILENO, &tinfostat);
      |                 ^~~~~~~~~
../text-utils/pg.c:1557:17: warning: nested extern declaration of ‘setupterm’ [-Wnested-externs]
[317/757] Compiling C object more.p/text-utils_more.c.o
FAILED: more.p/text-utils_more.c.o 
cc -Imore.p -I. -I.. -Iinclude -I../include -Ilibsmartcols -I../libsmartcols -I../libsmartcols/src -Ilibfdisk -I../libfdisk -I../libfdisk/src -I../libuuid/src -I../liblastlog2/src -Isys-utils -I../sys-utils -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -O0 -g -D_GNU_SOURCE -fsigned-char -include /var/home/jordan/Projects/util-linux/build/config.h -fno-common -Waddress-of-packed-member -Wdiscarded-qualifiers -Wformat-security -Wimplicit-function-declaration -Wmissing-declarations -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-missing-field-initializers -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wsign-compare -Wstrict-prototypes -Wtype-limits -Wuninitialized -Wunused-but-set-parameter -Wunused-but-set-variable -Wunused-parameter -Wunused-result -Wunused-variable -Wvla -Walloca -MD -MQ more.p/text-utils_more.c.o -MF more.p/text-utils_more.c.o.d -o more.p/text-utils_more.c.o -c ../text-utils/more.c
../text-utils/more.c: In function ‘checkf’:
../text-utils/more.c:459:25: error: implicit declaration of function ‘putp’; did you mean ‘puts’? [-Wimplicit-function-declaration]
  459 |                         putp(ctl->erase_line);
      |                         ^~~~
      |                         puts
../text-utils/more.c:459:25: warning: nested extern declaration of ‘putp’ [-Wnested-externs]
../text-utils/more.c: In function ‘more_exit’:
../text-utils/more.c:786:9: error: implicit declaration of function ‘del_curterm’ [-Wimplicit-function-declaration]
  786 |         del_curterm(cur_term);
      |         ^~~~~~~~~~~
../text-utils/more.c:786:9: warning: nested extern declaration of ‘del_curterm’ [-Wnested-externs]
../text-utils/more.c:786:21: error: ‘cur_term’ undeclared (first use in this function)
  786 |         del_curterm(cur_term);
      |                     ^~~~~~~~
../text-utils/more.c:786:21: note: each undeclared identifier is reported only once for each function it appears in
../text-utils/more.c: In function ‘initterm’:
../text-utils/more.c:2013:9: error: implicit declaration of function ‘setupterm’ [-Wimplicit-function-declaration]
 2013 |         setupterm(term, 1, &ret);
      |         ^~~~~~~~~
../text-utils/more.c:2013:9: warning: nested extern declaration of ‘setupterm’ [-Wnested-externs]
../text-utils/more.c:2019:39: error: implicit declaration of function ‘tigetnum’ [-Wimplicit-function-declaration]
 2019 |                 ctl->lines_per_page = tigetnum(TERM_LINES);
      |                                       ^~~~~~~~
../text-utils/more.c:2019:39: warning: nested extern declaration of ‘tigetnum’ [-Wnested-externs]
../text-utils/more.c:2027:43: error: implicit declaration of function ‘tigetflag’ [-Wimplicit-function-declaration]
 2027 |         if ((ctl->lines_per_page <= 0) || tigetflag(TERM_HARD_COPY)) {
      |                                           ^~~~~~~~~
../text-utils/more.c:2027:43: warning: nested extern declaration of ‘tigetflag’ [-Wnested-externs]
../text-utils/more.c:2040:27: error: implicit declaration of function ‘tigetstr’ [-Wimplicit-function-declaration]
 2040 |         ctl->erase_line = tigetstr(TERM_CLEAR_TO_LINE_END);
      |                           ^~~~~~~~
../text-utils/more.c:2040:27: warning: nested extern declaration of ‘tigetstr’ [-Wnested-externs]
../text-utils/more.c:2040:25: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2040 |         ctl->erase_line = tigetstr(TERM_CLEAR_TO_LINE_END);
      |                         ^
../text-utils/more.c:2041:20: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2041 |         ctl->clear = tigetstr(TERM_CLEAR);
      |                    ^
../text-utils/more.c:2042:29: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2042 |         if ((ctl->enter_std = tigetstr(TERM_STANDARD_MODE)) != NULL) {
      |                             ^
../text-utils/more.c:2043:31: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2043 |                 ctl->exit_std = tigetstr(TERM_EXIT_STANDARD_MODE);
      |                               ^
../text-utils/more.c:2048:21: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2048 |         cursor_addr = tigetstr(TERM_HOME);
      |                     ^
../text-utils/more.c:2050:29: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2050 |                 cursor_addr = tigetstr(TERM_CURSOR_ADDRESS);
      |                             ^
../text-utils/more.c:2052:39: error: implicit declaration of function ‘tparm’ [-Wimplicit-function-declaration]
 2052 |                         cursor_addr = tparm(cursor_addr, 0, 0);
      |                                       ^~~~~
../text-utils/more.c:2052:39: warning: nested extern declaration of ‘tparm’ [-Wnested-externs]
../text-utils/more.c:2052:37: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2052 |                         cursor_addr = tparm(cursor_addr, 0, 0);
      |                                     ^
../text-utils/more.c:2057:34: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2057 |         if ((ctl->move_line_down = tigetstr(TERM_LINE_DOWN)) == NULL)
      |                                  ^
../text-utils/more.c:2059:25: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2059 |         ctl->clear_rest = tigetstr(TERM_CLEAR_TO_SCREEN_END);
      |                         ^
../text-utils/more.c:2060:32: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2060 |         if ((ctl->backspace_ch = tigetstr(TERM_BACKSPACE)) == NULL)
      |                                ^
[319/757] Compiling C object test_more.p/text-utils_more.c.o
FAILED: test_more.p/text-utils_more.c.o 
cc -Itest_more.p -I. -I.. -Iinclude -I../include -Ilibsmartcols -I../libsmartcols -I../libsmartcols/src -Ilibfdisk -I../libfdisk -I../libfdisk/src -I../libuuid/src -I../liblastlog2/src -Isys-utils -I../sys-utils -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -O0 -g -D_GNU_SOURCE -fsigned-char -include /var/home/jordan/Projects/util-linux/build/config.h -fno-common -Waddress-of-packed-member -Wdiscarded-qualifiers -Wformat-security -Wimplicit-function-declaration -Wmissing-declarations -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-missing-field-initializers -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wsign-compare -Wstrict-prototypes -Wtype-limits -Wuninitialized -Wunused-but-set-parameter -Wunused-but-set-variable -Wunused-parameter -Wunused-result -Wunused-variable -Wvla -Walloca -DTEST_PROGRAM -MD -MQ test_more.p/text-utils_more.c.o -MF test_more.p/text-utils_more.c.o.d -o test_more.p/text-utils_more.c.o -c ../text-utils/more.c
../text-utils/more.c: In function ‘checkf’:
../text-utils/more.c:459:25: error: implicit declaration of function ‘putp’; did you mean ‘puts’? [-Wimplicit-function-declaration]
  459 |                         putp(ctl->erase_line);
      |                         ^~~~
      |                         puts
../text-utils/more.c:459:25: warning: nested extern declaration of ‘putp’ [-Wnested-externs]
../text-utils/more.c: In function ‘more_exit’:
../text-utils/more.c:786:9: error: implicit declaration of function ‘del_curterm’ [-Wimplicit-function-declaration]
  786 |         del_curterm(cur_term);
      |         ^~~~~~~~~~~
../text-utils/more.c:786:9: warning: nested extern declaration of ‘del_curterm’ [-Wnested-externs]
../text-utils/more.c:786:21: error: ‘cur_term’ undeclared (first use in this function)
  786 |         del_curterm(cur_term);
      |                     ^~~~~~~~
../text-utils/more.c:786:21: note: each undeclared identifier is reported only once for each function it appears in
../text-utils/more.c: In function ‘initterm’:
../text-utils/more.c:2013:9: error: implicit declaration of function ‘setupterm’ [-Wimplicit-function-declaration]
 2013 |         setupterm(term, 1, &ret);
      |         ^~~~~~~~~
../text-utils/more.c:2013:9: warning: nested extern declaration of ‘setupterm’ [-Wnested-externs]
../text-utils/more.c:2019:39: error: implicit declaration of function ‘tigetnum’ [-Wimplicit-function-declaration]
 2019 |                 ctl->lines_per_page = tigetnum(TERM_LINES);
      |                                       ^~~~~~~~
../text-utils/more.c:2019:39: warning: nested extern declaration of ‘tigetnum’ [-Wnested-externs]
../text-utils/more.c:2027:43: error: implicit declaration of function ‘tigetflag’ [-Wimplicit-function-declaration]
 2027 |         if ((ctl->lines_per_page <= 0) || tigetflag(TERM_HARD_COPY)) {
      |                                           ^~~~~~~~~
../text-utils/more.c:2027:43: warning: nested extern declaration of ‘tigetflag’ [-Wnested-externs]
../text-utils/more.c:2040:27: error: implicit declaration of function ‘tigetstr’ [-Wimplicit-function-declaration]
 2040 |         ctl->erase_line = tigetstr(TERM_CLEAR_TO_LINE_END);
      |                           ^~~~~~~~
../text-utils/more.c:2040:27: warning: nested extern declaration of ‘tigetstr’ [-Wnested-externs]
../text-utils/more.c:2040:25: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2040 |         ctl->erase_line = tigetstr(TERM_CLEAR_TO_LINE_END);
      |                         ^
../text-utils/more.c:2041:20: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2041 |         ctl->clear = tigetstr(TERM_CLEAR);
      |                    ^
../text-utils/more.c:2042:29: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2042 |         if ((ctl->enter_std = tigetstr(TERM_STANDARD_MODE)) != NULL) {
      |                             ^
../text-utils/more.c:2043:31: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2043 |                 ctl->exit_std = tigetstr(TERM_EXIT_STANDARD_MODE);
      |                               ^
../text-utils/more.c:2048:21: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2048 |         cursor_addr = tigetstr(TERM_HOME);
      |                     ^
../text-utils/more.c:2050:29: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2050 |                 cursor_addr = tigetstr(TERM_CURSOR_ADDRESS);
      |                             ^
../text-utils/more.c:2052:39: error: implicit declaration of function ‘tparm’ [-Wimplicit-function-declaration]
 2052 |                         cursor_addr = tparm(cursor_addr, 0, 0);
      |                                       ^~~~~
../text-utils/more.c:2052:39: warning: nested extern declaration of ‘tparm’ [-Wnested-externs]
../text-utils/more.c:2052:37: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2052 |                         cursor_addr = tparm(cursor_addr, 0, 0);
      |                                     ^
../text-utils/more.c:2057:34: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2057 |         if ((ctl->move_line_down = tigetstr(TERM_LINE_DOWN)) == NULL)
      |                                  ^
../text-utils/more.c:2059:25: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2059 |         ctl->clear_rest = tigetstr(TERM_CLEAR_TO_SCREEN_END);
      |                         ^
../text-utils/more.c:2060:32: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2060 |         if ((ctl->backspace_ch = tigetstr(TERM_BACKSPACE)) == NULL)
      |                                ^
[323/757] Compiling C object lsmem.p/sys-utils_lsmem.c.o
ninja: build stopped: subcommand failed.
```

--- a/meson.build
+++ b/meson.build
@@ -261,6 +261,7 @@ if lib_ncursesw.found()
 else
   lib_ncurses = dependency(
     'ncurses',
+    disabler : true,
     required : get_option('ncurses'))
   headers += ['ncurses.h',
               'term.h']

